controllers/openstackdataplanenode_controller.go

```diff
@@ -204,8 +204,15 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
 	// all setup tasks complete, mark SetupReadyCondition True
 	instance.Status.Conditions.MarkTrue(dataplanev1beta1.SetupReadyCondition, condition.ReadyMessage)
 
-	r.Log.Info("Node", "DeployStrategy", instance.Spec.DeployStrategy.Deploy, "Node.Namespace", instance.Namespace, "Node.Name", instance.Name)
-	if instance.Spec.DeployStrategy.Deploy {
```

Comment:
ditto, this triggering condition needs to be preserved

Reply:
Hmm, na, I don't like it. It's just very unnatural, it has no equivalence in Kubernetes - where you create the object and it doesn't rollout. So from a user perspective, this introduces an interface that increases the perceived complexity of our solution since we need to block normal Kubernetes workflows.

I think the challenge we have is that we have all of these nested CR's that can each be individually created. But then we can create them as part of the OpenStackDataplane CR as well. From a larger, outside of the scope of this change, I would challenge that we should even do that at all.

I'll have a think about how we can avoid triggering here, since I can see it creates two jobs for the node:
```
❯ oc get openstackansibleee
NAME                                                    NETWORKATTACHMENTS   STATUS   MESSAGE
dataplane-deployment-configure-network-edpm-compute                          True     AnsibleExecutionJob complete
dataplane-deployment-configure-network-edpm-compute-0                        True     AnsibleExecutionJob complete
dataplane-deployment-configure-os-edpm-compute                               False    AnsibleExecutionJob is running
dataplane-deployment-configure-os-edpm-compute-0                             False    AnsibleExecutionJob is running
dataplane-deployment-install-os-edpm-compute                                 True     AnsibleExecutionJob complete
dataplane-deployment-install-os-edpm-compute-0                               True     AnsibleExecutionJob complete
dataplane-deployment-validate-network-edpm-compute                           True     AnsibleExecutionJob complete
dataplane-deployment-validate-network-edpm-compute-0                         True     AnsibleExecutionJob complete
```

One created by the role:
```
❯ oc get openstackansibleee dataplane-deployment-configure-network-edpm-compute -o jsonpath='{.metadata.ownerReferences}' | jq .
[
  {
    "apiVersion": "dataplane.openstack.org/v1beta1",
    "blockOwnerDeletion": true,
    "controller": true,
    "kind": "OpenStackDataPlaneRole",
    "name": "edpm-compute",
    "uid": "9a8d6b55-fe97-4916-94eb-dbe88a3a5a33"
  }
]
```

And one by the node:
```
❯ oc get openstackansibleee dataplane-deployment-configure-network-edpm-compute-0 -o jsonpath='{.metadata.ownerReferences}' | jq .
[
  {
    "apiVersion": "dataplane.openstack.org/v1beta1",
    "blockOwnerDeletion": true,
    "controller": true,
    "kind": "OpenStackDataPlaneNode",
    "name": "edpm-compute-0",
    "uid": "a282f7a7-7ce5-48dc-87dd-120936de1a3f"
  }
]
```

Seems to do the same with `NovaExternalCompute` too, and complains about the resource being owned by another controller.

Reply:
`NovaExternalCompute` issue caused by this:

So the `OpenStackDataplaneRole` controller is trying to update it:
```
{"level":"error","ts":"2023-07-07T16:52:22+10:00","msg":"Reconciler error","controller":"openstackdataplanerole","controllerGroup":"dataplane.openstack.org","controllerKind":"OpenStackDataPlaneRole","OpenStackDataPlaneRole":{"name":"edpm-compute","namespace":"openstack-operators"},"namespace":"openstack-operators","name":"edpm-compute","reconcileID":"b64c7a8b-5bc2-4d43-b842-7eb01ac7a58c","error":"DeployNovaExternalCompute error:: Object openstack-operators/edpm-compute-0 is already owned by another OpenStackDataPlaneNode controller edpm-compute-0","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/brendanshephard/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/brendanshephard/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/Users/brendanshephard/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235"}
```

But it's owned by the `OpenStackDataplaneNode` controller:
```
❯ oc get novaexternalcompute -o yaml | yq '.items[].metadata.ownerReferences'
- apiVersion: dataplane.openstack.org/v1beta1
  blockOwnerDeletion: true
  controller: true
  kind: OpenStackDataPlaneNode
  name: edpm-compute-0
  uid: 7413790d-0453-462a-8749-b296d205fd1a
```

This is likely because all objects are essentially setting the equivalent of `deployStrategy.Deploy` now and we don't stop the execution of nodes while allowing the role. 

This is the summation of the issue caused by this removal, and something I will continue to work on.